Fix return type of TypeFactoryBase<c10::DynamicType>::get

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151850
* #151849
* #151810
* #151807
* #151806
* #151805
* #151804
* __->__ #151803
* #151802
* #151801

getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

swolchok committed:
Fix return type of TypeFactoryBase<c10::DynamicType>::get

getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

[ghstack-poisoned]

swolchok committed:
Update on "Fix return type of TypeFactoryBase<c10::DynamicType>::get"


getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

[ghstack-poisoned]

malfet reviewed (approved):
Sure, though I wonder if any of those should be technically BC-breaking

swolchok commented:
> Sure, though I wonder if any of those should be technically BC-breaking

This is specifically changing from a copy of a singleton shared_ptr to a reference to said singleton shared_ptr, so I don't think so.

swolchok committed:
Update on "Fix return type of TypeFactoryBase<c10::DynamicType>::get"


getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

[ghstack-poisoned]

swolchok committed:
rebase and minor fixes as requested on "Fix return type of TypeFactoryBase<c10::DynamicType>::get"


getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

[ghstack-poisoned]

swolchok committed:
Update on "Fix return type of TypeFactoryBase<c10::DynamicType>::get"


getBaseType() actually returns a reference. This was causing shared_ptr copies.

Differential Revision: [D73376717](https://our.internmc.facebook.com/intern/diff/D73376717/)

[ghstack-poisoned]
